File: DesignPatternsExercises/java/VisitorDesignPattern/Exercise/Customer.java
import java.util.ArrayList;

public class Customer implements Visitable {
    private String name;
    private ArrayList<__________(1)__________> orders = new ArrayList<>();

    public Customer(String name) {
        this.name = name;
    }

    @Override
    public void accept(Visitor visitor) {
        visitor.__________(2)__________;
        System.out.println("Orders:");
        for (Order order : orders)
            order.__________(3)__________;
        System.out.println();
    }

    public void addOrder(Order order) {
        orders.add(order);
    }

    public String getName() {
        return name;
    }
}

File: DesignPatternsExercises/ruby/visitor_design_pattern/exercise/customer.rb
require_relative 'visitable'

class Customer < Visitable
  attr_reader :name

  def __________(1)__________(name)
    @name = name
    __________(2)__________ = []
  end

  def __________(3)__________(visitor)
    visitor.__________(4)__________
    puts 'Orders:'
    @orders.each { |order| order.__________(5)__________ }
    puts
  end

  def add_order(order)
    @orders << order
  end
end
